core/src/main/java/com/redhat/lightblue/client/LightblueClient.java

```diff
@@ -17,6 +17,6 @@
 
     LightblueBulkDataResponse bulkData(AbstractDataBulkRequest<AbstractLightblueDataRequest> requests) throws LightblueException;
 
-	<T> T data(AbstractLightblueDataRequest lightblueRequest, Class<T> type) throws LightblueException;
+	<T> T data(LightblueRequest lightblueRequest, Class<T> type) throws LightblueException;
```

Comment:
I feel strongly that the data api should be locked down to AbstractLightblueDataRequest implementations. The method is not designed to work with other types of implementations.